language/move-prover/spec-lang/src/translate.rs

```diff
@@ -4025,6 +4087,12 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         }
         // Next treat this as a call to a global function.
         let (module_name, name) = self.parent.module_access_to_parts(maccess);
+
+        // Ignore assert statement.
+        if name == self.parent.parent.assert_symbol() {
```

Comment:
Guess we need to check for the qualified name which I think move_lang has created after expansion, otherwise every function the user calls assert shadowing the builtin assert will be also ignored. Not sure though how move_lang represents those calls.

Reply:
Move lang doesn't seem to allow users to shadow the builtin assert, based on my shallow experiment.